src/Symfony/Component/Console/CommandLoader/ContainerCommandLoader.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Symfony\Component\Console\CommandLoader;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Console\Exception\CommandNotFoundException;
+
+/**
+ * Loads commands from a PSR-11 container.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class ContainerCommandLoader implements CommandLoaderInterface
+{
+    private $container;
+    private $commandNames;
+
+    public function __construct(ContainerInterface $container, array $commandNames)
+    {
+        $this->container = $container;
+        $this->commandNames = $commandNames;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($name)
+    {
+        if (!in_array($name, $this->commandNames, true)) {
```

Comment:
I have no strong opinion on this, I looked at the existing and it seems duplicating has been preferred https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/ServiceLocator.php#L47.
However it means that if a sub class extend `has()` for some reasons, `get()` would still use the "default" way. Not sure if it's a good thing, perhaps someone else can has an hint?